### DISCUSSION TITLE
Follow up on the output for testing FLHIST

### DISCUSSION THREAD

**Discussion by cecilehannay** [NCAR/amwg_dev#230] [TF4: Diagnostics, output]
Here is what we are output for our runs. Please review carefully and let me know if you miss anything. 

@adamrher: I will add CO2 
@brianpm: adding COSP. Please let me know how to do it. Last time I tried, it didn't work. 
@PeterHjortLauritzen : please do you need the 'WV_..' in this run.
@tilmes: do you need all the variables below. 

```
fincl1 = 'ACTNI', 'ACTNL', 'ACTREI', 'ACTREL', 'AODDUST', 'AODVIS', 'AODVISdn','BURDENBC', 'BURDENDUST', 'BURDENPOM', 'BURDENSEASALT', 
'BURDENSO4', 'BURDENSOA', 'CAPE', 'CCN3', 'CDNUMC', 'CH4', 'CLDHGH', 'CLDICE', 'CLDLIQ', 'CLDLOW', 'CLDMED', 'CLDTOT', 'CLOUD', 'CMFMC_DP', 
'CT_H2O', 'DCQ', 'DQCORE', 'DTCOND', 'DTCORE', 'DTV', 'EVAPPREC', 'EVAPSNOW', 'FCTI', 'FCTL', 'FICE', 'FLDS', 'FLNS', 'FLNSC', 'FLNT', 'FLNTC', 'FLUT', 
'FREQZM', 'FSDS', 'FSDSC', 'FSNS', 'FSNSC', 'FSNT', 'FSNTC', 'FSNTOA', 'ICEFRAC', 'LANDFRAC', 'LHFLX', 'LWCF', 'MPDICE', 'MPDLIQ', 'MPDQ', 'MPDT', 
'OCNFRAC', 'OMEGA', 'OMEGA500', 'PBLH', 'PHIS', 'PINT', 'PMID', 'PRECC', 'PRECL', 'PRECSC', 'PRECSL', 'PRECT', 'PS', 'PSL', 'PTEQ', 'PTTEND', 'Q', 
'QFLX', 'QRL', 'QRS', 'QTGW', 'RCMTEND_CLUBB', 'RELHUM', 'RVMTEND_CLUBB', 'SHFLX', 'SOLIN', 'SST', 'STEND_CLUBB', 'SWCF', 
'T', 'TAUX', 'TAUY', 'TFIX', 'TGCLDIWP', 'TGCLDLWP', 'TMQ', 'TREFHT', 'TS', 'TTGW', 'U', 'U10', 'UBOT', 'UTGWORO', 'UTGW_TOTAL', 
'V', 'VBOT', 'VTGWORO', 'VTGW_TOTAL', 'WPRTP_CLUBB', 'WPTHLP_CLUBB', 'Z3', 'ZMDQ', 'ZMDT', 'N2O',

'WV_phBF:A','WL_phBF:A','WI_phBF:A','SE_phBF:A','KE_phBF:A',
'WV_phBP:A','WL_phBP:A','WI_phBP:A','SE_phBP:A','KE_phBP:A',
'WV_phAP:A','WL_phAP:A','WI_phAP:A','SE_phAP:A','KE_phAP:A',
'WV_phAM:A','WL_phAM:A','WI_phAM:A','SE_phAM:A','KE_phAM:A',

'AODVISdn', 'AQSO4_H2O2','AQSO4_O3',
'bc_a1', 'bc_a4', 'dst_a1', 'dst_a2', 'dst_a3', 'ncl_a1',
'ncl_a1', 'ncl_a2', 'ncl_a3', 'pom_a1', 'pom_a4', 'so4_a1', 'so4_a2', 'so4_a3',
'soa_a1', 'soa_a2', 'bc_c1', 'bc_c4', 'dst_c1', 'dst_c2', 'dst_c3', 'ncl_c1',
'ncl_c1', 'ncl_c2', 'ncl_c3', 'pom_c1', 'pom_c4', 'so4_c1', 'so4_c2', 'so4_c3',
'soa_c1', 'soa_c2', 'num_a1', 'num_a2', 'num_a3', 'num_a4', 'num_c1', 'num_c2',
'num_c3', 'num_c4', 'bc_a1SFWET', 'bc_a4SFWET', 'dst_a1SFWET', 'dst_a2SFWET', 'dst_a3SFWET', 'ncl_a1SFWET',
'ncl_a2SFWET', 'ncl_a3SFWET', 'pom_a1SFWET', 'pom_a4SFWET', 'so4_a1SFWET', 'so4_a2SFWET', 'so4_a3SFWET', 'soa_a1SFWET',
'soa_a2SFWET', 'bc_c1SFWET', 'bc_c4SFWET', 'dst_c1SFWET', 'dst_c2SFWET', 'dst_c3SFWET', 'ncl_c1SFWET', 'ncl_c2SFWET',
'ncl_c3SFWET', 'pom_c1SFWET', 'pom_c4SFWET', 'so4_c1SFWET', 'so4_c2SFWET', 'so4_c3SFWET', 'soa_c1SFWET', 'soa_c2SFWET',
'bc_a1DDF', 'bc_a4DDF', 'dst_a1DDF', 'dst_a2DDF', 'dst_a3DDF', 'ncl_a1DDF', 'ncl_a2DDF', 'ncl_a3DDF',
'pom_a1DDF', 'pom_a4DDF', 'so4_a1DDF', 'so4_a2DDF', 'so4_a3DDF', 'soa_a1DDF', 'soa_a2DDF', 'bc_c1DDF',
'bc_c4DDF', 'dst_c1DDF', 'dst_c2DDF', 'dst_c3DDF', 'ncl_c1DDF', 'ncl_c2DDF', 'ncl_c3DDF', 'pom_c1DDF',
'pom_c4DDF', 'so4_c1DDF', 'so4_c2DDF', 'so4_c3DDF', 'soa_c1DDF', 'soa_c2DDF',
'num_a1DDF', 'num_a2DDF',
'num_a3DDF', 'num_a4DDF', 'num_c1DDF', 'num_c2DDF', 'num_c3DDF', 'num_c4DDF', 'bc_a4_CLXF', 'pom_a4_CLXF',
'so4_a1_CLXF', 'so4_a2_CLXF', 'num_a1_CLXF', 'num_a2_CLXF', 'SFbc_a4', 'SFpom_a4', 'SFso4_a1', 'SFso4_a2',
'SFnum_a1', 'SFnum_a2', 'SFnum_a3', 'so4_a1_sfgaex1', 'so4_a2_sfgaex1', 'so4_a3_sfgaex1', 'soa_a1_sfgaex1', 'soa_a2_sfgaex1',
'so4_a2_sfnnuc1', 'so4_c1AQH2SO4', 'so4_c2AQH2SO4', 'so4_c3AQH2SO4', 'so4_c1AQSO4', 'so4_c2AQSO4', 'so4_c3AQSO4', 'SFdst_a1',
'SFdst_a2', 'SFdst_a3', 'SFncl_a1', 'SFncl_a2', 'SFncl_a3', 'soa_a1_CHML', 'soa_a1_CHML', 'so4_a1_CHMP',
'so4_a2_CHMP', 'so4_a3_CHMP', 'soa_a1_CHMP', 'soa_a1_CHMP', 'H2SO4_sfnnuc1',
'num_a2_sfnnuc1'
```
```
fincl2 = 'OMEGA', 'PMID', 'PS', 'Q', 'QRL', 'QRS', 'T', 'TROP_P', 'TROP_T', 'U', 'V', 'Z3'
```
```
fincl4 =  'PRECC','PRECL'
```

**Comment by PeterHjortLauritzen** [comment]
Reply to "@PeterHjortLauritzen : please do you need the 'WV_..' in this run." 

No

**Reply by cecilehannay** [reply]
Thanks. I will remove it for this run.

**Comment by brianpm** [comment]
For turning on COSP, it is just:
./xmlchange --append CAM_CONFIG_OPTS=-cosp

That should turn on COSP and add a bunch of default monthly output.

**Reply by adamrher** [reply]
should we do a quick test to make sure empty_htapes=T doesn't override the cosp diagnostics?

**Reply by cecilehannay** [reply]
Hi @brianpm,

We tried to do this (add ``-cosp`` to ``CAM_CONFIG_OPTS``) at Andrew's request a few months ago in: 
``b.cesm3_cam041_mom.B1850MOM.ne30_L58_t061.004``

We were also using in the namelist:
```
cosp_passive = .true.
docosp = .true. 
```
But we had issues and we turn it off as we needed to keep going. I am not sure crash were due to the extra namelist flag. 

But I will give it a try.

**Reply by brianpm** [reply]
First, empty_htapes likely WILL clobber the COSP outputs, so I should send a list. (When do you need it?)

Second, give it a try. I'm not sure why any of the COSP stuff would be broken, but if it is, I can help look into it. It's possible the namelist flags could be faulty, so you could just exclude those (they *should not* make a difference since those flags would turn on all of the simulators AFAIK).

**Reply by brianpm** [reply]
Here are my current top priorities for COSP output:

- CLD_MISR
- FISCCP1_COSP
- CLD_CAL
- CLD_MISR
- CLDTOT_CAL
- CLDHGH_CAL
- CLDMED_CAL
- CLDLOW_CAL
- CLMODIS (<- forgot this one at first, edited to add)

From these, I think I can do all the diagnostics that I've been doing for my CAM4-5-6-E3SMv1 comparison, which is currently submitted to ESS (submitted version here: https://www.dropbox.com/s/0ggdsv68z58610k/medeiros_shaw_kay_davis_ess-SUBMIT.pdf?dl=0)

**Comment by cecilehannay** [comment]
Diags for the previous run are here: 
https://webext.cgd.ucar.edu/FLTHIST/f.cam6_3_101.FLTHIST_v0a.ne30.001/atm/f.cam6_3_101.FLTHIST_v0a.ne30.001_1979_1985_vs_f.cam6_3_100.FWscHIST.ne30_L58.001_1979_1985/index.html

Nothing crazy in the LW.

**Comment by tilmes** [comment]
I am thinking about a light version of the aerosol output. The list above is not all needed for the general budget, so I reduced it to what is really needed. Sorry I did not sort it out earlier.
'AQSO4_H2O2','AQSO4_O3', 'bc_a1', 'bc_a4', 'dst_a1', 'dst_a2', 'dst_a3', 'ncl_a1',
'ncl_a1', 'ncl_a2', 'ncl_a3', 'pom_a1', 'pom_a4', 'so4_a1', 'so4_a2', 'so4_a3',
'soa_a1', 'num_a1', 'num_a2', 'num_a3', 'num_a4',
 'bc_a1SFWET', 'bc_a4SFWET', 'dst_a1SFWET', 'dst_a2SFWET', 'dst_a3SFWET', 'ncl_a1SFWET',
'ncl_a2SFWET', 'ncl_a3SFWET', 'pom_a1SFWET', 'pom_a4SFWET', 'so4_a1SFWET', 'so4_a2SFWET', 'so4_a3SFWET', 'soa_a1SFWET',
'soa_a2SFWET', 'bc_c1SFWET', 'bc_c4SFWET', 'dst_c1SFWET', 'dst_c2SFWET', 'dst_c3SFWET', 'ncl_c1SFWET', 'ncl_c2SFWET',
'ncl_c3SFWET', 'pom_c1SFWET', 'pom_c4SFWET', 'so4_c1SFWET', 'so4_c2SFWET', 'so4_c3SFWET', 'soa_c1SFWET', 'soa_c2SFWET',
'bc_a1DDF', 'bc_a4DDF', 'dst_a1DDF', 'dst_a2DDF', 'dst_a3DDF', 'ncl_a1DDF', 'ncl_a2DDF', 'ncl_a3DDF',
'pom_a1DDF', 'pom_a4DDF', 'so4_a1DDF', 'so4_a2DDF', 'so4_a3DDF', 'soa_a1DDF', 'soa_a2DDF', 
'so4_a1_CLXF', 'so4_a2_CLXF', 'SFbc_a4', 'SFpom_a4', 'SFso4_a1', 'SFso4_a2',
 'so4_a1_sfgaex1', 'so4_a2_sfgaex1', 'so4_a3_sfgaex1', 'soa_a1_sfgaex1', 'soa_a2_sfgaex1',
'SFdst_a1','SFdst_a2', 'SFdst_a3', 'SFncl_a1', 'SFncl_a2', 'SFncl_a3',
'num_a2_sfnnuc1'

In addition, we could also combine, say bc_a1 and bc_a4, but then we don't have the information on those parts that interact with clouds and those that don't, so I am not doing it here. We could, however combine other variables and still do the budget. This could be the aerosol light list, but it also requires some namelist additions to sum up the specific variables.  Would this be a more reasonable list?

'AQSO4_H2O2','AQSO4_O3', 'bc_a1','bc_a4', 'dst_a2', 'dst_a3', 'ncl_a1',
'ncl_a1', 'ncl_a2', 'ncl_a3', 'pom_a1', 'pom_a4', 'so4_a1', 'so4_a2', 'so4_a3',
'soa_a1', 'num_a1', 'num_a2', 'num_a3', 'num_a4',
'bc_SFWET', 'dst_SFWET','ncl_SFWET','pom_SFWET','so4_SFWET','soa_SFWET'
'bc_DDF', 'dst_DDF,'ncl_DDF,'pom_DDF','so4_DDF','soa_DDF'
'SFbc_a4','SFpom_a4' 'SFdst', 'SFso4','so4_sfgaex1','soa_sfgex1'
'num_a2_sfnnuc1'

**Comment by dan800** [comment]
@tilmes Can you put together the necessary namelist definitions to create this lighter aerosol list? It's needed for https://github.com/NCAR/amwg_dev/issues/233

**Comment by tilmes** [comment]
@cecilehannay 
 I just learned that my short namelist does actually not work, so I guess we have to just not add the longer one as often for now.
